src/utils/readFile.ts

```diff
@@ -0,0 +1,35 @@
+import * as RNFS from 'react-native-fs'
+import Logger from 'src/utils/Logger'
+
+/**
+ * Reads and returns a portion of the input file.
+ */
+const readChunk = async (file: string, length: number, position: number) => {
+  try {
+    return await RNFS.read(file, length, position, 'utf8')
+  } catch (error) {
+    Logger.error('Logger@readChunk', 'readChunk error', error as Error)
```

Comment:
why swallow the error (instead of bubbling it up)?

Reply:
Couldn't you just re-throw the error here?  You have the log message and you'll break out of your do while.  It seems odd to swallow the error in low-level file reading code.  Shouldn't higher-level code handle the error and decide what to do with it?

Reply:
Also, is `.error` the right level here?  Seems or should it be `.warn`?  `.error` implies a series problem we didn't expect and we should investigate immediately, but it looks like we might expect this?